app/src/main/java/com/andela/helpmebuy/adapters/ReceivedRequestAdapter.java

```diff
@@ -42,30 +42,29 @@ public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
     public void onBindViewHolder(final CustomViewHolder holder, final int position) {
         DateManager dateManager = new DateManager();
         PurchaseRequest request = purchaseRequests.get(position);
-        holder.sender.setText(request.getSendersFullName());
         String items = getItemsDescription(request);
+        String date = dateManager.formatTime(request.getDateMillis());
+        holder.sender.setText(request.getSendersFullName());
         holder.description.setText(items);
-        holder.date.setText(dateManager.formatTime(request.getDateMillis()));
+        holder.date.setText(date);
     }
 
     public String getItemsDescription(PurchaseRequest request){
         String itemName = "";
         ArrayList<PurchaseItem> items = request.getPurchaseList();
-        if(items.size() == 1) {
-            return itemName + items.get(0).getItemName();
+        if(items.size() > 0) {
```

Comment:
NITPICK: Not sure if other if blocks would run, since what you have would always be greater than 0. We only want 3 things to happen, if the item is 1, more than one but less than 3 and more than 3. Check your conditions around that.